src/lib/storage.js

```diff
@@ -503,11 +505,21 @@ class Storage implements IStorageHandler {
           return callback(null, packageInfo);
         }
 
-        self.localStorage.updateVersions(name, packageInfo, function(err, packageJsonLocal: Package) {
+        self.localStorage.updateVersions(name, packageInfo, async (err, packageJsonLocal: Package) => {
           if (err) {
             return callback(err);
           }
-          return callback(null, packageJsonLocal, upLinksErrors);
+          const filterErrors = [];
+          for (let i = 0; i < self.filters.length; ++i) {
```

Comment:
I think `for ... of` is more suitable here or `Promise.all([])`
https://lavrton.com/javascript-loops-how-to-handle-async-await-6252dd3c795/

Reply:
`for of` is fine, but I should comment `Promise.all()` would be a bug as we want these to run in series

Reply:
You are right, ignore the `Promise.all()`